frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/NextBranchCheckoutAction.java

```diff
@@ -0,0 +1,38 @@
+package com.virtuslab.gitmachete.frontend.actions.contextmenu;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseCheckoutAction;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+
+@ExtensionMethod(GitMacheteBundle.class)
+public class NextBranchCheckoutAction extends BaseCheckoutAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  protected @Nullable String getSubjectBranchName(AnActionEvent anActionEvent) {
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+    if (gitRepository != null) {
+      val localBranches = gitRepository.getBranches().getLocalBranches().iterator();
+      while (localBranches.hasNext()) {
+        val nextBranch = localBranches.next();
+        if (nextBranch.getName() == currentBranchName) {
```

Comment:
Whoops `equals`!

Reply:
Oh, Interning Checker captured that 💪🏻

Reply:
Interning Checker is a bit PITA (with a all that esoteric annotations like `@UsesObjectEquals` but captures incorrectly used `==` which is awesome

Reply:
Note that it doesn't trigger locally: https://github.com/VirtusLab/git-machete-intellij-plugin/blob/master/CONTRIBUTING.md#build -> `-PrunAllCheckers`